packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -496,177 +496,187 @@ export default class TableResize implements EditorPlugin {
                 this.isRTL ? rect.left : rect.right,
                 !this.isRTL
             );
-            this.nextCellsToResize = vtable.getCellsWithBorder(
-                this.isRTL ? rect.left : rect.right,
-                this.isRTL
-            );
         }
-
         this.startResizeCells(e);
     };
 
     private startResizeCells(e: MouseEvent) {
         const doc = this.editor.getDocument();
         doc.addEventListener('mousemove', this.frameAnimateResizeCells, true);
         doc.addEventListener('mouseup', this.endResizeCells, true);
+        setEmptyTableCells(this.currentTable);
     }
 
     private frameAnimateResizeCells = (e: MouseEvent) => {
         this.editor.runAsync(() => this.resizeCells(e));
     };
 
-    private resizeCells = (e: MouseEvent) => {
-        this.setTableResizer(null);
-        if (this.resizingState === ResizeState.None) {
-            return;
-        } else if (this.resizingState === ResizeState.Both) {
-            let rect = normalizeRect(this.currentTable.getBoundingClientRect());
-            let vtable = new VTable(this.currentTable);
+    private resizeTable = (mouseX: number, mouseY: number) => {
+        let vtable = new VTable(this.currentTable);
 
-            let currentBorder: number = parseFloat(
-                this.currentTable.getAttribute(
-                    this.isRTL ? 'currentLeftBorder' : 'currentRightBorder'
-                )
-            );
-            const tableBottomBorder: number = parseFloat(
-                this.currentTable.getAttribute('currentBottomBorder')
-            );
-            const ratioX =
-                1.0 +
-                (this.isRTL
-                    ? (currentBorder - e.pageX) / (rect.right - currentBorder)
-                    : (e.pageX - currentBorder) / (currentBorder - rect.left));
-            const ratioY = 1.0 + (e.pageY - tableBottomBorder) / (tableBottomBorder - rect.top);
-
-            const shouldResizeX = Math.abs(ratioX - 1.0) > 1e-3;
-            const shouldResizeY = Math.abs(ratioY - 1.0) > 1e-3;
-            if (shouldResizeX || shouldResizeY) {
-                for (let i = 0; i < vtable.cells.length; i++) {
-                    for (let j = 0; j < vtable.cells[i].length; j++) {
-                        const cell = vtable.cells[i][j];
-                        if (cell.td) {
-                            if (shouldResizeX) {
-                                const originalWidth: number = cell.td.style.width
+        if (!this.isReadyForResizingTable) {
+            preProcessTable(vtable.table, ResizeState.Both);
+            this.isReadyForResizingTable = true;
+        }
+        let rect = normalizeRect(vtable.table.getBoundingClientRect());
+        let currentBorder: number = parseFloat(
+            vtable.table.getAttribute(this.isRTL ? 'currentLeftBorder' : 'currentRightBorder')
```

Comment:
Where is "currentLeftBorder" removed after resize?

Reply:
And since this is only used internally, I prefer make it simpler to "currentVerticalBorder", so no need to do RTL check everywhere.

Reply:
But actually why we need to write this result to table? Can't we just remember it as a local variable or member field?

Reply:
we don't remove it after resize, instead, we update it with the new border position based on the new resized position so that we can use set the correct value for the next resizing. I don't use as a local variable, considering the fact that we might have multiple tables, one single local variable won't be enough, so I save it as an attribute.

Reply:
used the 'currentVerticalBorder'

Reply:
fixed by using a new strategy.